test/test_octave_action.py

```diff
@@ -2,42 +2,25 @@
 #
 # Copyright (c) Bo Peng and the University of Texas MD Anderson Cancer Center
 # Distributed under the terms of the 3-clause BSD License.
+
 import os
 import shutil
-import unittest
-
-from sos.parser import SoS_Script
-from sos.targets import file_target
-from sos.utils import env
-from sos.workflow_executor import Base_Executor
-
-
-class TestActions(unittest.TestCase):
-
-    def setUp(self):
-        env.reset()
-        self.temp_files = []
 
-    def tearDown(self):
-        for f in self.temp_files:
-            file_target(f).unlink()
+import pytest
 
-    @unittest.skipIf(not shutil.which('octave'), 'Octave not installed')
-    def test_octave(self):
-        '''Test action octave'''
-        if os.path.isfile('octave_example.txt'):
-            os.remove('octave_example.txt')
-        script = SoS_Script(r'''
-[0]
-octave:
-    f = fopen("octave_example.txt", "w")
-    fprintf(f, "A, B, C, D\n")
-    fclose(f)
-''')
-        wf = script.workflow()
-        Base_Executor(wf).run()
-        self.assertTrue(os.path.isfile('octave_example.txt'))
+from sos import execute_workflow
 
 
-if __name__ == '__main__':
-    unittest.main()
+@pytest.mark.skipif(not shutil.which('octave'), reason = 'Octave not installed')
+def test_octave():
+    '''Test action octave'''
+    if os.path.isfile('octave_example.txt'):
+        os.remove('octave_example.txt')
```

Comment:
Is this an instance where `os.remove` should be left as is?

Reply:
No. It should be replaced with the fixture. In your case you removed the entire `os.remove` without replacing it with the fixture.